2005-02-14  Jeff Johnston <jjohnstn@redhat.com>

	* top.c (gdb_readline_wrapper): Ensure terminal is gdb's before calling
	readline.

Index: gdb-6.5/gdb/top.c
===================================================================
--- gdb-6.5.orig/gdb/top.c	2006-03-29 19:53:33.000000000 -0300
+++ gdb-6.5/gdb/top.c	2006-07-07 02:29:42.000000000 -0300
@@ -728,6 +728,14 @@ gdb_readline_wrapper (char *prompt)
       after_char_processing_hook = NULL;
     }
 
+  /* Before calling readline, ensure we have the terminal.  If we don't
+     have the terminal and call readline, we risk the possibility of
+     gdb being thrown into the background.  This problem occurs when
+     we attach to a background process on the same terminal the background
+     process was started from and then perform some action which requires
+     a page break prompt.  */
+  terminal_ours ();
+
   return readline (prompt);
 }
 
